dashboard/app/batch_coverage.go

```diff
@@ -151,3 +153,71 @@ func nsDataAvailable(ctx context.Context, ns string) ([]coveragedb.TimePeriod, [
 	}
 	return periods, recordsCount, nil
 }
+
+// Abandoned DB cleanup is expensive.
+// Every namespace creates min 3 sessions every day (yesterday, today, this month).
+// It means every day we deprecate 3 sessions records.
+// Every session is ~10k file records. Plus some mutations in the index.
+// One transaction allows to delete up to 80k rows.
+// To clean up everything once/week we have to garbage 7 * 3 * count(namespaces) at least.
+func handleBatchCoverageClean(w http.ResponseWriter, r *http.Request) {
+	paramSessions := r.FormValue("sessions")
+	if paramSessions == "" {
+		paramSessions = "3" // one day, one namespace means 3 records
+	}
+	maxSessionsToDel, err := strconv.Atoi(paramSessions)
+	if err != nil {
+		w.WriteHeader(http.StatusBadRequest)
+		w.Write([]byte("failed to parse 'sessions', integer expected"))
+		return
+	}
+	ctx := context.Background()
+	for i := 0; i < maxSessionsToDel; i++ {
+		deletedRows, err := deleteGarbageSession(ctx)
+		if err != nil {
+			errMsg := fmt.Sprintf("failed to cleanCoverageDB: %s", err.Error())
+			log.Errorf(ctx, "%s", errMsg)
+			w.Write([]byte(errMsg))
+			return
+		}
+		w.Write([]byte(fmt.Sprintf("deleteGarbageSession -> -%d rows\n", deletedRows)))
+	}
+}
+
+// deleteGarbageSession generates approximately 20k mutations (10k in files and 10k in index)
+// Spanner limit for every transaction is 80k mutations.
+// It means we can delete up to 4 sessions data at once.
+// Let's keep it simple and delete only 1 session at once.
+//
+// deleteGarbageSession returns the deleted rows count.
+// (0, nil) means there is no data to delete.
+func deleteGarbageSession(ctx context.Context) (int64, error) {
```

Comment:
That might better belong to `pkg/coveragedb`.

Reply:
done